pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -631,15 +631,20 @@ protected void startStatsUpdater(int statsUpdateInitialDelayInSecs, int statsUpd
     }
 
     protected void startDeduplicationSnapshotMonitor() {
+        // We do not know whether users will enable deduplication on namespace level/topic level or not, so keep this
+        // scheduled task runs.
         int interval = pulsar().getConfiguration().getBrokerDeduplicationSnapshotFrequencyInSeconds();
-        if (interval > 0 && pulsar().getConfiguration().isBrokerDeduplicationEnabled()) {
+        if (interval > 0) {
             this.deduplicationSnapshotMonitor = OrderedScheduler.newSchedulerBuilder()
                     .name("deduplication-snapshot-monitor")
                     .numThreads(1)
                     .build();
             deduplicationSnapshotMonitor.scheduleAtFixedRate(() -> forEachTopic(
                     Topic::checkDeduplicationSnapshot)
                     , interval, interval, TimeUnit.SECONDS);
+        } else {
+            throw new IllegalArgumentException("The config brokerDeduplicationSnapshotFrequencyInSeconds should be"
```

Comment:
Wouldn't this prevent the broker from starting? I guess that is fine, but it would be great to have a way to document this in release notes as an intentional breaking change.

Reply:
It will prevent the broker from starting.

Reply:
Created a separate PR https://github.com/apache/pulsar/pull/22480 to do this change